sdk/python/tests/unit/test_feature_server.py

```diff
@@ -209,6 +225,244 @@ def test_materialize_request_model():
     assert req2.end_ts == "2021-01-02T00:00:00"
 
 
+def _enable_offline_batching_config(
+    fs, enabled: bool = True, batch_size: int = 1, batch_interval_seconds: int = 60
+):
+    """
+        Attach a minimal feature_server.offline_push_batching config
+    to a mocked FeatureStore.
+    """
+    if not hasattr(fs, "config") or fs.config is None:
+        fs.config = SimpleNamespace()
+
+    if not hasattr(fs.config, "feature_server") or fs.config.feature_server is None:
+        fs.config.feature_server = SimpleNamespace()
+
+    fs.config.feature_server.offline_push_batching_enabled = enabled
+    fs.config.feature_server.offline_push_batching_batch_size = batch_size
+    fs.config.feature_server.offline_push_batching_batch_interval_seconds = (
+        batch_interval_seconds
+    )
+
+
+def push_body_many(push_mode=PushMode.ONLINE, count: int = 2, id_start: int = 100):
+    """Build a push body with multiple entities."""
+    driver_ids = list(range(id_start, id_start + count))
+    lats = [float(i) for i in driver_ids]
+    longs = [str(lat) for lat in lats]
+    event_ts = [str(_utc_now()) for _ in range(count)]
+    created_ts = [str(_utc_now()) for _ in range(count)]
+
+    return {
+        "push_source_name": "driver_locations_push",
+        "df": {
+            "driver_lat": lats,
+            "driver_long": longs,
+            "driver_id": driver_ids,
+            "event_timestamp": event_ts,
+            "created_timestamp": created_ts,
+        },
+        "to": push_mode.name.lower(),
+    }
+
+
+@pytest.mark.parametrize("online_write", [True, False])
+@pytest.mark.parametrize("batching_enabled", [True, False])
+@pytest.mark.parametrize(
+    "push_mode",
+    [PushMode.ONLINE, PushMode.OFFLINE, PushMode.ONLINE_AND_OFFLINE],
+)
+def test_push_batched_matrix(
+    online_write, batching_enabled, push_mode, mock_fs_factory
+):
+    """
+    Matrix over:
+      - online_write ∈ {True, False}
+      - batching_enabled ∈ {True, False}
+      - push_mode ∈ {ONLINE, OFFLINE, ONLINE_AND_OFFLINE}
+
+    Asserts:
+      - which of fs.push / fs.push_async are called
+      - how many times
+      - with which `to` values
+
+    For batching_enabled=True, batch_size=1 ensures immediate flush of offline part.
+    """
+    fs = mock_fs_factory(online_write=online_write)
+
+    _enable_offline_batching_config(
+        fs,
+        enabled=batching_enabled,
+        batch_size=1,  # flush immediately on a single offline request
+        batch_interval_seconds=60,
+    )
+
+    client = TestClient(get_app(fs))
+
+    # use a multi-row payload to ensure we test non-trivial dfs
+    resp = client.post("/push", json=push_body_many(push_mode, count=2, id_start=100))
+    assert resp.status_code == 200

```

Comment:
optional but I think it's good to return 202 when batching is enabled and offline writes are involved